fhevm/tee_interpreter.go

```diff
@@ -0,0 +1,195 @@
+
+	resultHash := resultCt.GetHash()
+	logger.Info(fmt.Sprintf("%s success", op), "lhs", lhs.hash().Hex(), "rhs", rhs.hash().Hex(), "result", resultHash.Hex())
+	return resultHash[:], nil
+}
+
+func extract2Operands(op string, environment EVMEnvironment, input []byte, runSpan trace.Span) (*tee.TeePlaintext, *tee.TeePlaintext, *verifiedCiphertext, *verifiedCiphertext,	error) {
+	input = input[:minInt(65, len(input))]
+
+	logger := environment.GetLogger()
+
+	lhs, rhs, err := get2VerifiedOperands(environment, input)
+	otelDescribeOperands(runSpan, encryptedOperand(*lhs), encryptedOperand(*rhs))
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s inputs not verified", op), "err", err, "input", hex.EncodeToString(input))
+		return nil, nil, nil, nil, err
+	}
+	if lhs.fheUintType() != rhs.fheUintType() {
+		logger.Error(fmt.Sprintf("%s operand type mismatch", op), "lhs", lhs.fheUintType(), "rhs", rhs.fheUintType())
+		return nil, nil, nil, nil, errors.New("operand type mismatch")
+	}
+
+	lp, err := tee.Decrypt(lhs.ciphertext)
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s failed", op), "err", err)
+		return nil, nil, lhs, rhs, err
+	}
+
+	rp, err := tee.Decrypt(rhs.ciphertext)
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s failed", op), "err", err)
+		return nil, nil, lhs, rhs, err
+	}
+
+	return &lp, &rp, lhs, rhs, nil
+}
+
+func extract3Operands(op string, environment EVMEnvironment, input []byte, runSpan trace.Span) (*tee.TeePlaintext, *tee.TeePlaintext, *tee.TeePlaintext, *verifiedCiphertext, *verifiedCiphertext, *verifiedCiphertext, error) {
+	input = input[:minInt(96, len(input))]
+
+	logger := environment.GetLogger()
+
+	fhs, shs, ths, err := get3VerifiedOperands(environment, input)
+	otelDescribeOperands(runSpan, encryptedOperand(*fhs), encryptedOperand(*shs), encryptedOperand(*ths))
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s inputs not verified", op), "err", err, "input", hex.EncodeToString(input))
+		return nil, nil, nil, nil, nil, nil, err
+	}
+	if shs.fheUintType() != ths.fheUintType() {
+		logger.Error(fmt.Sprintf("%s operand type mismatch", op), "shs", shs.fheUintType(), "ths", ths.fheUintType())
+		return nil, nil, nil, nil, nil, nil, errors.New("operand type mismatch")
+	}
+
+	fp, err := tee.Decrypt(fhs.ciphertext)
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s failed", op), "err", err)
+		return nil, nil, nil, fhs, shs, ths, err
+	}
+
+	sp, err := tee.Decrypt(shs.ciphertext)
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s failed", op), "err", err)
+		return nil, nil, nil, fhs, shs, ths, err
+	}
+
+	tp, err := tee.Decrypt(ths.ciphertext)
+	if err != nil {
+		logger.Error(fmt.Sprintf("%s failed", op), "err", err)
+		return nil, nil, nil, fhs, shs, ths, err
+	}
+
+	return &fp, &sp, &tp, fhs, shs, ths, nil
+}
+
+// marshalTfheType converts a any to a byte slice
+func marshalTfheType(value any, typ tfhe.FheUintType) ([]byte, error) {
+	var resultBz []byte
+
+	switch any(value).(type) {
```

Comment:
```suggestion
	switch value := any(value).(type) {
```

In this case, below, you don't need to do value.(uint64) or value.(bool)